lib/ansible/module_utils/ordereddict.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c), Matt Martz <sivel@ansible.com> 2018
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from collections import MutableMapping
+
+from ansible.module_utils.six.moves import map as _imap
+from operator import eq as _eq
+
+try:
+    from collections import KeysView, ValuesView, ItemsView
+except ImportError:
+    from collections.abc import KeysView, ValuesView, ItemsView
+
+try:
+    from ansible.module_utils.six.moves._thread import get_ident as _get_ident
+except ImportError:
+    from ansible.module_utils.six.moves._dummy_thread import get_ident as _get_ident
+
+
+class OrderedDict(dict):
```

Comment:
@sivel From the rest of the code, it looks like you probably have a reason for inheriting from dict instead of from MutableMapping.  Probably need a comment to explain that, though.

Reply:
Honestly, it wasn't a choice that I specifically made.  This implementation came from https://github.com/ActiveState/code/tree/3b27230f418b714bc9a0f897cb8ea189c3515e99/recipes/Python/576693_Ordered_Dictionary_for_Py24

I only made some modifications to utilize some of our compat code.

Reply:
We probably want to inherit from MutableMapping instead of dict, then.

Reply:
what would be the advantages/dis for either case?

Reply:
https://github.com/xonsh/xonsh/pull/125#discussion_r26907490